metrics/metrics_collector.go

```diff
@@ -0,0 +1,155 @@
+package metrics
+
+import (
+	"fmt"
+	"log"
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type metricCollector struct {
+	counters   []prometheus.Counter
+	gauges     []prometheus.Gauge
+	histograms []prometheus.Histogram
+	summarys   []prometheus.Summary
+	interval   time.Duration
+}
+
+var (
+	promRegistry = prometheus.NewRegistry() // local Registry so we don't get Go metrics, etc.
+)
+
+func newMetricCollector() metricCollector {
+	return metricCollector{}
+}
+
+func (mc *metricCollector) updateCounter() {
+	for _, c := range mc.counters {
+		c.Add(rand.Float64())
+	}
+
+}
+
+func (mc *metricCollector) updateGauge() {
+	for _, c := range mc.gauges {
+		c.Add(rand.Float64())
+	}
+}
+
+func (mc *metricCollector) updateHistogram() {
+	for idx := 0; idx < len(mc.histograms); idx++ {
+		lowerBound := math.Mod(rand.Float64(), 1)
+		increment := math.Mod(rand.Float64(), 0.05)
+		for i := lowerBound; i < 1; i += increment {
+			mc.histograms[idx].Observe(i)
+
+		}
+	}
+
+}
+func (mc *metricCollector) updateSummary() {
+	for idx := 0; idx < len(mc.summarys); idx++ {
+		lowerBound := math.Mod(rand.Float64(), 1)
+		increment := math.Mod(rand.Float64(), 0.05)
+		for i := lowerBound; i < 1; i += increment {
+			mc.summarys[idx].Observe(i)
+
+		}
+
+	}
+}
+
+func updateLoop(update func(), delay time.Duration) {
+	delay = delay * time.Second
```

Comment:
This conversion should probably happen when the `delay` is initialized since the CLI runner is what determines that its configuration is denominated in seconds.  This function shouldn't care what duration it receives.